pkg/core/resources/apis/mesh/mesh_validator.go

```diff
@@ -165,8 +167,12 @@ func validateDatadog(cfgStr *structpb.Struct) validators.ValidationError {
 		verr.AddViolation("address", "cannot be empty")
 	}
 
-	if cfg.Port > 0xFFFF || cfg.Port < 1 {
-		verr.AddViolation("port", "must be in the range 1 to 65535")
+	if !strings.HasPrefix(cfg.Address, "unix:") {
+		if elms := strings.Split(cfg.Address, ":"); len(elms) < 2 {
```

Comment:
Will incorporate whatever we land on in the PR which just addresses this format change.